app/components/StatefulLink/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+*
+* StatefulLink
+*
+*  Parameters >>
+*  - [state] State that want to pass to another page
+*  - [path] Navigation Path
+*
+*  Sample Code >>
+*  - <StatefulLink state={{ a: 1 }} path="/mall">title</StatefulLink>
+*
+*  Ways to get state >>
+*  - this.props.history.location.state === { a: 1 }
```

Comment:
```
this.props.location.state
```
instead of this.history. will cause many issues later when implementing, and required `withRouter` connection from 'react-router-dom'